nixos/modules/services/x11/redshift.nix

```diff
@@ -113,22 +148,42 @@ in {
 
     systemd.user.services.redshift =
     let
-      providerString = if lcfg.provider == "manual"
-        then "${toString lcfg.latitude}:${toString lcfg.longitude}"
-        else lcfg.provider;
+      # This should be appended last in the config file since it introduces
+      # its own section "[manual]"
+      locationString = if lcfg.provider == "manual"
+        then ''
+          [manual]
+          lat=${toString lcfg.latitude}
+          lon=${toString lcfg.longitude}
+        ''
+        else "";
+      fadeString = if cfg.fade then "1" else "0";
+      sectionString = if strings.hasInfix "gammastep" cfg.executable
+        # https://gitlab.com/chinstrap/gammastep/-/commit/1608ed61154cc652b087e85c4ce6125643e76e2f
+        then "[general]"
+        else "[redshift]";
+      configFile = pkgs.writeText "redshift.conf" ''
+        ${sectionString}
+        temp-day=${toString cfg.temperature.day}
+        temp-night=${toString cfg.temperature.night}
+        fade=${fadeString}
+        location-provider=${lcfg.provider}
+        brightness-day=${toString cfg.brightness.day}
+        brightness-night=${toString cfg.brightness.night}
+        gamma-day=${toString cfg.gamma.day}
+        gamma-night=${toString cfg.gamma.night}
+
+        ${cfg.extraOptions}
+
+        ${locationString}
```

Comment:
Probably there is a better way to do it where it wouldn't leave blank spaces if the option is blank, but anyway this doesn't affect the result so I didn't really bother.

Reply:
This is solved if you use the `toINI` generator.